src/components/learner-credit-management/BudgetDetailAssignments.jsx

```diff
@@ -26,20 +29,26 @@ const BudgetDetailAssignments = ({
     assignmentConfigurationUUID,
     enterpriseId,
   });
-
+  
   if (!isTopDownAssignmentEnabled || !isAssignableBudget) {
     return null;
   }
-
+  
   if (!hasContentAssignments && hasSpentTransactions) {
     return (
       <AssignMoreCoursesEmptyStateMinimal />
     );
   }
-
+  
+  useEffect(() => {
```

Comment:
As is, this `useEffect` is breaking the Rules of Hooks (important ESLint error).

<img width="1104" alt="image" src="https://github.com/openedx/frontend-app-admin-portal/assets/2828721/f4b1c617-b885-49fd-b15a-fb467f1a3a79">